query-parser/src/parse.rs

```diff
@@ -70,6 +69,130 @@ macro_rules! matches_plain_symbol {
     }
 }
 
+trait FromValue<T> {
```

Comment:
Hmm.  Is there a reason not to use `TryFrom`, or to implement `T : From<edn::Value>`?

There are many ways to skin this cat in Rust, so I'm not too concerned about getting this just right at this moment, but I'd like to know if you have done some of this thinking.